### DISCUSSION TITLE
I can't make it print level lower than info

### DISCUSSION THREAD

**Discussion by Cyberavater** [klogging/klogging#222] [Q&A] [answered]
```kotlin

suspend fun main(): Unit = coroutineScope {
    loggingConfiguration {
        ANSI_CONSOLE()
        minDirectLogLevel(Level.DEBUG)
    }
    val logger = logger("main")
    logger.debug {
        "Debug msg"
    }

    logger.info {
        "Info msg"
    }
}

```

it only prints 
```
20:07:54.643211600  INFO [                main] :        Configuration : Configuration set using the DSL with append=false
20:07:54.706432600  INFO [                main] :                 main : Info msg
```

how do I make it print debug?

**Comment by mjstrasser** [comment, marked as answer]
Hi @Cyberavater, thanks for your question.

You have stumbled upon an undocumented ‘feature’ of Klogging (i.e. incomplete documentation)! The simple [ANSI_CONSOLE](https://github.com/klogging/klogging/blob/main/klogging/src/commonMain/kotlin/io/klogging/config/Defaults.kt#L78) configuration sets the minimum logging level to `INFO`.

Sorry for that inconvenience. You can use:

```kotlin
loggingConfiguration {
    sink("console", STDOUT_ANSI)
    logging { fromMinLevel(DEBUG) { toSink("console") } }
    minDirectLogLevel(Level.DEBUG)
}
```

I will update the documentation soon.

**Reply by Cyberavater** [reply]
Thanks this works, but why are we setting the level to debug twice?
` sink("console", STDOUT_ANSI)` and `minDirectLogLevel(Level.DEBUG)`
What's the difference between them?

**Reply by mjstrasser** [reply]
`fromMinLogLevel()` sets the minimum level at which events are sent to the specified sink. [Documentation of this and similar DSL functions](https://klogging.io/docs/concepts/direct-logging) provides more detail.

`minDirectLogLevel()` sets the minimum level at which events are processed directly, rather than asynchronously. [Direct logging](https://klogging.io/docs/concepts/direct-logging) provides more detail. You need this in a simple main program like your example, or else [you won’t see the log messages you expect](https://klogging.io/docs/get-started#i-didnt-see-any-logs).

**Reply by Cyberavater** [reply]
Thanks for the feedback.

**Comment by CoreyKaylor** [comment]
This 'feature' is a bug IMO. It should respect the log levels as configured. If sinks should override a log level, that should be expressed in the configuration, not a default for a given sink. This will trip up every person using the library.

**Reply by mjstrasser** [reply]
The problem is that the names do not explain the whole story: that they are complete configurations, including minimum logging level.

Release 0.11.1 of Klogging updates that name to `ANSI_INFO` (deprecating the old one) which is clearer.

[The documentation](https://klogging.io/docs/configuration/built-ins/#complete-configurations) of this and other complete configurations is clearer too, I hope.
